metrics/firebase/docs/features/builds_aggregation/01_firestore_builds_aggregation_design.md

```diff
@@ -0,0 +1,176 @@
+
+### Firestore
+
+The section contains information about the main purposes of new collections, their document structures with field descriptions, and a set of security rules.
+
+### `build_days`
+
+> Explain the main purpose of the collection.
+
+The first collection we should create is the `build_days`. It holds builds grouped by the `status` and `day`. Each status contains the count of builds, created per day. 
+
+#### Document Structure
+
+> Explain the structure of the documents under this collection.
+
+We should produce a composite document identifier that consists of a project's id and a day this aggregation document belongs to. This identifier we can use to easily update a value of the `build_days` document.
+
+The collection's document has the following structure:
+
+```json
+{
+  "projectId": String,
+  "successful": number,
+  "failed": number,
+  "unknown": number,
+  "inProgress": number,
+  "totalDuration": number,
+  "day": timestamp,
+}
+```
+
+Let's take a closer look at the document's fields:
+
+| Field | Description |
+| --- | --- |
+| `projectId`   | An identifier of the project, this aggregation relates to. |
+| `successful` | A count of builds with a `successful` status. |
+| `failed` | A count of builds with a `failed` status. |
+| `unknown` | A count of builds with an `unknown` status. |
+| `inProgress` | A count of builds with an `inProgress` status. |
+| `totalDuration` | A total builds duration. |
+| `day`   | A timestamp that represents the day start this aggregation belongs to. |
+
+#### Security Rules
+
+> Explain the Firestore Security Rules for this collection.
+
+Every authenticated user can read the documents from the `build_days` collection, but no one can write to this collection. Let's consider the security rules for this collection in more details:
+
+- Access-related rules:
+  - not authenticated users **cannot** read, create, update, delete this document;
+  - authenticated users **can** read and **cannot** create, update, delete this document.
+
+### `tasks`
+
+> Explain the main purpose of the collection.
+
+There may be situations where creating or updating the `build_days` collection fails. If it happens, we should store the required for builds aggregation fields to the `data` field of the specific collection - `tasks` to process this data later.
+
+#### Document Structure
+
+> Explain the structure of the documents under this collection.
+
+The collection's document has the following structure:
+
+```json
+{
+  "code": String,
+  "data": Map,
+  "context": String,
+  "createdAt": timestamp,
+}
+```
+
+Let's take a closer look at the document's fields:
+
+| Field | Description |
+| --- | --- |
+| `code`   | A string that identifies the task. |
```

Comment:
```suggestion
| `code`   | A string that identifies the task to perform. |
```
Does it make sense since this field does not uniquely identify the `task` document?